contracts/farm/LiquidityMining.sol

```diff
@@ -582,21 +605,31 @@ contract LiquidityMining is ILiquidityMining, LiquidityMiningStorage {
         bool _isCheckingBalance
     ) internal {
         uint256 userAccumulatedReward = _user.accumulatedReward;
+        /// @dev get unlock immediate percent of the pool token.
+        uint256 _unlockedImmediatelyPercent = _getPoolTokenUnlockedImmediatelyPercent(_poolToken);
 
         /// @dev Transfer if enough SOV balance on this LM contract.
         uint256 balance = SOV.balanceOf(address(this));
         if (balance >= userAccumulatedReward) {
             totalUsersBalance = totalUsersBalance.sub(userAccumulatedReward);
             _user.accumulatedReward = 0;
 
-            /// @dev Instead of transferring the reward to the LP (user),
-            ///   deposit it into lockedSOV vault contract, but first
+            /// @dev If _unlockedImmediatelyPercent is 100%, transfer the reward to the LP (user).
+            ///   else, deposit it into lockedSOV vault contract, but first
             ///   SOV deposit must be approved to move the SOV tokens
             ///   from this LM contract into the lockedSOV vault.
-            require(SOV.approve(address(lockedSOV), userAccumulatedReward), "Approve failed");
-            lockedSOV.deposit(_userAddress, userAccumulatedReward, unlockedImmediatelyPercent);
+            if (_unlockedImmediatelyPercent == 10000) {
+                SOV.transfer(_userAddress, userAccumulatedReward);
+            } else {
+                require(SOV.approve(address(lockedSOV), userAccumulatedReward), "Approve failed");
+                lockedSOV.deposit(
+                    _userAddress,
+                    userAccumulatedReward,
+                    _unlockedImmediatelyPercent
+                );
+            }
 
-            if (_isStakingTokens) {
+            if (_isStakingTokens && _unlockedImmediatelyPercent < 10000) {
```

Comment:
can be moved up to `else` condition then `_unlockedImmediatelyPercent < 10000` can be removed

Reply:
updated